internal/lockservice/simpleLockService.go

```diff
@@ -42,41 +54,56 @@ type SimpleLockService struct {
 	lockMap *SafeLockMap
 }
 
-var _ Descriptors = (*SimpleDescriptor)(nil)
+var _ Descriptors = (*LockDescriptor)(nil)
 
-// SimpleDescriptor implements the Descriptors interface.
+// ObjectDescriptor describes the object that is subjected to
+// lock operations.
+type ObjectDescriptor struct {
```

Comment:
Why can't LockDescriptor be used in the cache-related functions?

Reply:
Check #21

Reply:
Oh right. I'd forgotten why we did this.